src/anyio/_backends/_asyncio.py

```diff
@@ -2729,14 +2746,14 @@ async def wait_readable(cls, obj: FileDescriptorLike) -> None:
 
         read_events[obj] = event
         try:
+            await cls.checkpoint()
```

Comment:
I think this checkpoint is redundant because wait_readable won't callback without an event loop cycle so it runs selector.select()